vortex-duckdb/src/convert/array/varbinview.rs

```diff
@@ -0,0 +1,107 @@
+use std::ffi::c_char;
+
+use duckdb::vtab::arrow::WritableVector;
+use itertools::Itertools;
+use vortex_array::arrays::{Inlined, VarBinViewArray};
+use vortex_error::VortexResult;
+
+use crate::ToDuckDB;
+use crate::buffer::{
+    AssignBufferToVec, FFIDuckDBBuffer, FFIDuckDBBufferInternal, NewCppVectorBuffer,
+};
+
+// This is the C++ string view struct
+// private:
+// 	union {
+// 		struct {
+// 			uint32_t length;
+// 			char prefix[4];
+// 			char *ptr;
+// 		} pointer;
+// 		struct {
+// 			uint32_t length;
+// 			char inlined[12];
+// 		} inlined;
+// 	} value;
+// };
+
+#[derive(Clone, Copy)]
+#[repr(C, align(16))]
+pub union PtrBinaryView {
+    // Numeric representation. This is logically `u128`, but we split it into the high and low
+    // bits to preserve the alignment.
+    le_bytes: [u8; 16],
+
+    // Inlined representation: strings <= 12 bytes
+    inlined: Inlined,
+
+    // Reference type: strings > 12 bytes.
+    _ref: PtrRef,
+}
+
+#[derive(Clone, Copy, Debug)]
+#[repr(C, align(8))]
+pub struct PtrRef {
+    size: u32,
+    prefix: [u8; 4],
+    ptr: *const c_char,
+}
+
+impl ToDuckDB for VarBinViewArray {
+    fn to_duckdb(&self, chunk: &mut dyn WritableVector) -> VortexResult<()> {
+        let buffers = self.buffers();
+        let mut buffer_used = vec![false; buffers.len()];
+
+        let views: Vec<PtrBinaryView> = self
```

Comment:
This function is quite long and dense. Might be worth considering breaking it up.